packages/mobile/src/send/utils.ts

```diff
@@ -105,3 +118,48 @@ export function showLimitReachedError(
 
   return showError(ErrorMessages.PAYMENT_LIMIT_REACHED, ALERT_BANNER_DURATION, translationParams)
 }
+
+export function* handleSendPaymentData(data: UriData, cachedRecipient?: RecipientWithContact) {
+  const recipient: RecipientWithQrCode = {
+    kind: RecipientKind.QrCode,
+    address: data.address,
+    displayId: data.e164PhoneNumber,
+    displayName: data.displayName || cachedRecipient?.displayName || 'anonymous',
+    phoneNumberLabel: cachedRecipient?.phoneNumberLabel,
+    thumbnailPath: cachedRecipient?.thumbnailPath,
+    contactId: cachedRecipient?.contactId,
+  }
+  yield put(storeLatestInRecents(recipient))
+
+  if (data.currencyCode) {
+    yield put(selectPreferredCurrency(data.currencyCode as LocalCurrencyCode))
+    yield take([Actions.FETCH_CURRENT_RATE_SUCCESS, Actions.FETCH_CURRENT_RATE_FAILURE])
+  }
+
+  if (data.amount) {
+    const exchangeRate = yield select(getLocalCurrencyExchangeRate)
+    const amount = convertLocalAmountToDollars(data.amount, exchangeRate)
+    if (!amount) {
+      Logger.warn('handleSendPaymentData', 'null amount')
+      return
+    }
+    const transactionData: TransactionDataInput = {
+      recipient,
+      amount,
+      reason: data.comment,
```

Comment:
Does this reason get set as a comment here? https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/send/SendConfirmation.tsx#L147